test/typescript/index.ts

```diff
@@ -0,0 +1,43 @@
+import React from "react";
+
+const equal = require("../..");
+
+type DeepObject = {
+  any: any;
```

Comment:
two things...
- i wouldn't use `any` as a key name as that's gonna get confusing.
- i wouldn't use `any` type as it might actually allow compilation where it should have failed